RecoParticleFlow/PFClusterProducer/interface/PFHGCalRecHitCreator.h

```diff
@@ -58,8 +59,15 @@ template <typename DET,PFLayer::Layer Layer,ForwardSubdetector subdet>
 	double energy = hgrh.energy();
 	double time = hgrh.time();	
 	
-	const FlatTrd *thisCell = 
-	  static_cast<const FlatTrd*>(hgcGeo.getGeometry(detid));	
+	const CaloCellGeometry *thisCell = nullptr;
+	
+	if( DetId::Forward == detid.det() ) {
+	  const auto* hgGeom = static_cast<const HGCalGeometry*>(geom->getSubdetectorGeometry(detid.det(),detid.subdetId()));
+	  thisCell = static_cast<const FlatTrd*>(hgGeom->getGeometry(detid));
```

Comment:
What you say is true but there's actually an underlying problem. The way this was rewritten by Vincenzo is wrong. getGeometry() for the HGCal returns a whole wafer and not a cell. So the positions of the hits are being assigned incorrectly. Nominally HGCal hits are assigned their position dynamically from the geometry.

However, this means putting the position back in the PFRecHit since a pointer cannot be stored, that or having the PFRecHit store a function pointer to how to get the position or something.